src/inverse/SIA2D/gradient.jl

```diff
@@ -120,13 +132,13 @@ function SIA2D_grad_batch!(θ, simulation::FunctionalInversion)
                 ℓ += Δt[j-1]*ℓi
 
                 ### Custom VJP to compute the adjoint
-                λ_∂f∂H, dH_H = VJP_λ_∂SIA∂H(simulation.parameters.UDE.grad.VJP_method, λ[j], H[j], θ, simulation, t₀)
+                λ_∂f∂H, dH_H = VJP_λ_∂SIA∂H(simulation.parameters.UDE.grad.VJP_method, λ[j], H[j], θ, simulation, tj)
```

Comment:
Was this a bug in the original code? The fact that this was calling the time t0 rather that then current time? I think this probably had no effect before, since none of the dynamics depended of time before (with MB, this is no longer true)